optuna/visualization/_contour.py

```diff
@@ -90,14 +91,21 @@ def _get_contour_plot(study: Study, params: Optional[List[str]] = None) -> "go.F
 
     padding_ratio = 0.05
     param_values_range = {}
+    update_category_axes = {}
     for p_name in sorted_params:
         values = [t.params[p_name] for t in trials if p_name in t.params]
-        max_value = max(values)
+
         min_value = min(values)
+        max_value = max(values)
+
         if _is_log_scale(trials, p_name):
             padding = (math.log10(max_value) - math.log10(min_value)) * padding_ratio
             min_value = math.pow(10, math.log10(min_value) - padding)
             max_value = math.pow(10, math.log10(max_value) + padding)
+
+        elif _is_categorical(trials, p_name):
+            update_category_axes[p_name] = any([str(v).isnumeric() for v in set(values)])
```

Comment:
For future readers, how about leaving a brief comment about why `update_category_axes` is needed if a categorical parameter contains numeric like values?